site-src/geps/gep-1424.md

```diff
@@ -0,0 +1,384 @@
+    * When an `HTTPBackendRef` is invalid, 500 status codes MUST be returned for requests that would have otherwise been routed to an invalid backend.
+        * This matches existing behavior as defined in [`HTTPRouteRule`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPRouteRule).
+* SHOULD set a status field on `HTTPRoute` to show if the routing configuration has been applied to the mesh.
+* MAY assume that a mesh implements "transparent proxy" functionality to redirect calls to the Kubernetes DNS address for a `Service` through mesh routing rules.
+    * For all namespaces configured by a cluster operator in the scope of a mesh deployment.
+* SHOULD NOT require reconfiguring existing `HTTPRoutes` for North/South Gateway configuration.
+
+## Non-Goals
+
+* Supporting "egress" use cases, which is currently a deferred goal, including:
+    * Defining how "consumer" traffic rules which could override routing for service upstreams only within the local scope of a namespace or service might be configured.
+    * Redirecting calls from arbitrary custom domains to an in-cluster service.
+* Defining how multiple `Services` or `EndpointSlices` representing instances of a single "logical" service should present an identity for AuthN/AuthZ or be associated with each other beyond routing rules.
+* Defining how AuthZ should be implemented to secure East/West traffic between services.
+* Defining how [Policy Attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/) would bind to `HTTPRoute`, services or a mesh.
+* Defining how `HTTPRoutes` configured for East/West service mesh traffic management might integrate with North/South `Gateways`.
+    * This is a bit tricky in that it's effectively a form of delegation as described in [GEP-1058: Route Inclusion and Delegation](https://github.com/kubernetes-sigs/gateway-api/pull/1085), and is planned to be explored in a future GEP.
+* Handling East/West traffic outside the cluster (VMs, etc).
+
+## Introduction
+
+It is proposed that an application owner should configure traffic rules for a mesh service by configuring an `HTTPRoute` with a Kubernetes `Service` resource as a `parentRef`.
+
+This approach is dependent on both the "frontend" role of the Kubernetes `Service` resource as defined in [GEP-1324: Service Mesh in Gateway API](https://gateway-api.sigs.k8s.io/geps/gep-1324/#service) when used as a `parentRef` and the "backend" role of `Service` when used as a `backendRef`. It would use the Kubernetes DNS name to match traffic for meshes implementing "transparent proxy" functionality, but the `backendRef` endpoints would ultimately be used for the canonical IP address(es) to which traffic should be redirected by rules defined in this HTTPRoute. This approach leverages the existing points of extensibility within the Gateway API spec, and would not require introducing any API changes or new resources, only defining expected behavior.
```

Comment:
```suggestion
This approach is dependent on both the "frontend" role of the Kubernetes `Service` resource as defined in [GEP-1324: Service Mesh in Gateway API](https://gateway-api.sigs.k8s.io/geps/gep-1324/#service) when used as a `parentRef` and the "backend" role of `Service` when used as a `backendRef`. It would use the Kubernetes service name to match traffic for meshes implementing "transparent proxy" functionality, but the `backendRef` endpoints would ultimately be used for the canonical IP address(es) to which traffic should be redirected by rules defined in this HTTPRoute. This approach leverages the existing points of extensibility within the Gateway API spec, and would not require introducing any API changes or new resources, only defining expected behavior.
```

Without further explanation, "DNS" is confusing here.

What we are describing is "Calls to service foo". For `Service`, that is identified by the ClusterIP of `foo`. For 99.99% of users, they access this by doing a DNS lookup for `foo` (or variants); but the API itself has no direct relation to DNS, IMO.

Might be worth calling this out (unless we do later on) to clarify, since it is subtle